### PR TITLE
added login_as field searchable

### PR DESCRIPTION
**Pull request by adil-uddin**
##### SUMMARY
https://dimagi-dev.atlassian.net/browse/USH-150

##### FEATURE FLAG
RESTRICT_LOGIN_AS

##### RISK ASSESSMENT / QA PLAN
No QA. Low-risk change, has test coverage, isolated behind feature flag.

##### PRODUCT DESCRIPTION
Makes the `login_as_user` user data field searchable from mobile workers page search box.

### CODE REVIEW

**Comment by orangejenny** [comment]
Code looks fine, I'm going to hold off on merging until you've had a chance to dig some more into why/how this works.

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
Test failures look real. Speaking of tests, it would also be good to add test coverage for this PR.

```
root: ERROR: You need to define UserESFake.OR. Here's a start
    def OR(self, ...):
        # called with args ({'nested': {'path': 'user_data_es', 'query': {'bool': {'filter': ({'term': {'user_data_es.key': 'login_as_user'}}, {'match': {'user_data_es.value': {'query': 'no-match', 'fuzziness': 0}}})}}}}, {'query_string': {'query': '*no* *match*', 'default_operator': 'AND', 'fields': ['base_username', 'last_name', 'first_name']}}) and kwargs {}
        return self._filtered(lambda doc: ...)
```

File: corehq/apps/users/analytics.py [hunk 1]
```diff
@@ -71,8 +71,8 @@ def get_search_users_in_domain_es_query(domain, search_string, limit, offset):
     default_search_fields = ["base_username", "last_name", "first_name"]
     return (UserES()
             .domain(domain)
-            .OR(users.metadata('login_as_user', search_string),
-                queries.search_string_query(search_string, default_search_fields))
+            .filter(filters.OR(users.metadata('login_as_user', search_string),
```
Comment (esoergel) [inline corehq/apps/users/analytics.py:74]: what's with the change from `.OR()` to `.filter(filters.OR())`?  The former is syntactic sugar for the latter - they should produce identical queries.

Comment (adil-uddin) [reply corehq/apps/users/analytics.py:74]: Ha, please ignore that was just my lame attempt at trying to trick the test case :P

**Comment by esoergel** [comment]
Thought I just had on this PR - I think the query logic should be behind the `RESTRICT_LOGIN_AS` feature flag, since it's not applicable to any other projects.  Incidentally, that might let you get away without changing so much in the test infrastructure, since you'd only need the `OR` filter when the feature flag is enabled - so that could be tested in isolation.

**Comment by adil-uddin** [comment]
@esoergel @orangejenny looks like the build passed. Can you please review.

**Review by orangejenny — APPROVED** [review]

**Pull request merged**